lib/jobs.js

```diff
@@ -35,19 +37,26 @@ async function unzip(config) {
         const zipBuffer = new AdmZip(zipFile.body);
         const zipEntries = zipBuffer.getEntries();
 
-        const uploadFiles = zipEntries.map(async zipEntry => {
-            const fileName = zipEntry.entryName;
-            const file = Buffer.from(zipEntry.getData());
-
-            return store.putArtifact(config.buildId, config.token, fileName, file);
-        });
-
         if (PARALLEL_UPLOAD_LIMIT <= 0) {
-            await Promise.all(uploadFiles);
+            await Promise.all(
+                zipEntries.map(async zipEntry => {
+                    const fileName = zipEntry.entryName;
+                    const file = Buffer.from(zipEntry.getData());
+
+                    return store.putArtifact(config.buildId, config.token, fileName, file);
+                })
+            );
         } else {
-            while (uploadFiles.length > 0) {
+            while (zipEntries.length > 0) {
                 // eslint-disable-next-line no-await-in-loop
-                await Promise.all(uploadFiles.splice(0, PARALLEL_UPLOAD_LIMIT));
+                await Promise.all(
+                    zipEntries.splice(0, PARALLEL_UPLOAD_LIMIT).map(async zipEntry => {
+                        const fileName = zipEntry.entryName;
+                        const file = Buffer.from(zipEntry.getData());
+
+                        return store.putArtifact(config.buildId, config.token, fileName, file);
```

Comment:
These functions passed to `map` function are same, so can you defines the function with appropriate name and passed it to each `map` to reduce redundant code?

Reply:
Thx.
I fixed it.